app/feature/feature-odyssey/src/main/kotlin/com/hedvig/android/feature/odyssey/step/summary/ClaimSummaryDestination.kt

```diff
@@ -103,16 +137,112 @@ private fun ClaimSummaryScreen(
     ) {
       Column(sideSpacingModifier.fillMaxWidth()) {
         for ((left, right) in detailPairs) {
-          Row {
-            Text(text = left, modifier = Modifier.weight(1f))
-            Text(text = right, modifier = Modifier.weight(1f), textAlign = TextAlign.End)
+          HorizontalItemsWithMaximumSpaceTaken(
+            startSlot = {
+              Text(text = left)
+            },
+            endSlot = {
+              Text(text = right, modifier = Modifier.weight(1f), textAlign = TextAlign.End)
+            },
+          )
+        }
+      }
+    }
+    Spacer(Modifier.height(24.dp))
+    Text(
+      stringResource(R.string.claim_status_detail_uploaded_files_info_title),
+      sideSpacingModifier,
+    )
+    Spacer(Modifier.height(8.dp))
+    when (uiState.claimSummaryInfoUiState.submittedContent) {
+      is SubmittedContent.Audio -> {
+        val signedAudioUrl = SignedAudioUrl.fromSignedAudioUrlString(
+          uiState.claimSummaryInfoUiState.submittedContent.audioURL,
+        )
+        Column(sideSpacingModifier) {
+          val audioPlayer = rememberAudioPlayer(
+            playableAudioSource = PlayableAudioSource.RemoteUrl(signedAudioUrl),
+          )
+          HedvigAudioPlayer(audioPlayer = audioPlayer)
+        }
+      }
+
+      else -> {}
+    }
+    Spacer(Modifier.height(8.dp))
+    val fileListSize = uiState.claimSummaryInfoUiState.files.size
+    val calculatedHeight =
+      if (fileListSize == 0) {
+        0.dp
+      } else if (fileListSize <= 3) {
+        130.dp
+      } else if (fileListSize <= 6) {
+        260.dp
+      } else {
+        300.dp
+      }
+    LazyVerticalGrid(
+      modifier = Modifier.height(calculatedHeight),
```

Comment:
Oh sorry I just realized I didn't mention that part. I was thinking we can just use `FilesLazyVerticalGrid` directly, not a custom one built for this place here. I feel like the way it's written it should be possible to re-use it, could you give that a try?

And then since that grid is more dynamic regarding how many columns it gives for the grid, we can instead not do any kind of height calculation ourselves here, and give it a sane default and let it be.
```suggestion
    LazyVerticalGrid(
      modifier = Modifier.height(300.dp),
```
I know this will mean on 1 file it will take a bit too much, and that on multiple files it will only show some files completely, but since we are in this scrollable-column inside a scrollable-column situation we gotta make some compromise I think.

It is also possible that I am thinking about this wrong, I think it might be a good idea for us to look at it together a bit so that I can also see how this visually looks like when we run it.

Reply:
I have changed it to FilesLazyVerticalGrid in the last commit, but I'm actually not sure it is a better decision than this custom grid here. Or maybe some other thing altogether? I still left checking for an empty list, since for this case it really looks bad with empty 300.dp and it's easy to fix, but otherwise it's not quite possible to make it look good with with for example only 2 files and fixed 300.dp height (see screenshot, the button is hidden below with no reason). I looked up how it's done on AddFilesScreen, but the situation is a bit different there:)

![Screenshot_20240229_174953_Hedvig Dev](https://github.com/HedvigInsurance/android/assets/90948269/8c6f10a4-4ccf-40ea-94f8-2d0439e1d207)

Reply:
Yeahh I agree it's looking odd. Let's look at it together soon.